frontend/src/components/ProjectPicker.vue

```diff
@@ -0,0 +1,191 @@
+<template>
+  <div class="container">
+    <h1 class="title">Project Picker</h1>
+
+    <div class="new-projects">
+      <router-link :to="{ name: 'NewProject' }">
+        <button class="primary-button">Create New Project</button>
+      </router-link>
+    </div>
+
+    <!-- Display projects if available -->
+    <div class="projects-grid">
+      <div v-for="project in projects" :key="project.id" class="project-card">
+        <div class="delete-button" @click.stop="deleteProject(project.id)">
+          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="white" width="18px" height="18px">
+            <path d="M0 0h24v24H0z" fill="none"/>
+            <path d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"/>
+          </svg>
+        </div>
+        <router-link :to="{ name: 'ProjectEditor', params: { id: project.id } }" class="project-link">
+          <img :src="project.thumbnail_path" loading="lazy" :alt="'Project ' + project.id" class="project-thumbnail"/>
+          <h3 class="project-title">{{ project.id }}</h3>
+        </router-link>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted } from "vue";
+import axios from "axios";
+
+const BACKEND_URL = import.meta.env.VITE_BACKEND_URL;
+console.log('Backend URL:', BACKEND_URL);
+
+let projects = ref<Array<ProjectResponse>>([]);
+
+const getProjects = async () => {
+  try {
+    const res = await axios.get(BACKEND_URL + "/projects");
+    projects.value = res.data.projects;
+    console.log('Projects:', projects.value);
+  } catch (error) {
+    console.error('Error fetching projects:', error);
+  }
+}
+
+const deleteProject = async (projectId: string) => {
+  try {
+    projects.value = projects.value.filter(project => project.id !== projectId);
+    await axios.delete(`${BACKEND_URL}/projects/${projectId}`);
+    console.log(`Project ${projectId} deleted successfully`);
+  } catch (error) {
+    console.error(`Error deleting project ${projectId}:`, error);
+    // If the delete request fails, add the project back to the list
+    await getProjects();
+  }
+}
+
+onMounted(() => {
+  getProjects();
+});
+
+interface ProjectResponse {
+  id: string,
+  thumbnail_path: string,
+}
+</script>
+<style scoped>
+body {
```

Comment:
Bei dir ist ein Component = ein "Router View". Heißt, dein Router tauscht eigentlich nur einen Component aus und der ist der gesamte Body. Daher hier okay, aber ich strukturiere meine SPAs gerne etwas anders:

Ich hab immer gerne nen extra Ordner für "Views" die dann vom Router ausgetauscht werden.
Diese Views sind zwar auch "Components", sind aber irgendwie größer. Fassen also mehrere kleinere Components zusammen und bilden einen "Abschnitt".

In meinem "components" Ordner leben dann wirklich nur Sachen, die wiederverwendet werden könnten.
Nen View ist ja immer nur einmal da und daher _eigentlich_ kein Component sondern ne Component Gruppe.
Aber das ist nur die Art und Weise wie ich das gerne Strukturiere (und ich halte mich auch nicht immer dran, wenn nen View nicht viel können muss, kommt die Funktionalität halt direkt da rein ohne nen Extra Component mit den Funktionen).

Hier tut das jetzt nicht weh und ist okay alles als component zu pflegen weil die Dateien noch recht klein sind, aber wenns größer wird, könnte das irgendwann unübersichtlich werden.